### PR TITLE
Shuffle peers when broadcasting messages in the mock p2p backend

### PR DESCRIPTION
**Pull request by stanislav-tkach**
Closes https://github.com/mintlayer/mintlayer-core/issues/511.

### CODE REVIEW

File: p2p/src/net/mock/backend.rs [hunk 1]
```diff
@@ -252,13 +253,23 @@ where
...
+            .iter()
+            .map(|(id, p)| {
+                p.tx.send(subscription.clone()).inspect_err(move |e| {
+                    log::error!("Failed to send subscription to {id:?} peer: {e:?}")
+                })
             })
-        }))
-        .await;
+            .collect();
+        futures.shuffle(&mut rand::thread_rng());
```
Comment (azarovh) [inline p2p/src/net/mock/backend.rs:270]: Shouldn't `crypto::random::make_pseudo_rng()` be used instead?

**Review by azarovh — APPROVED** [review]

File: p2p/src/net/mock/backend.rs (hunk 1 above)
Comment (stanislav-tkach) [reply p2p/src/net/mock/backend.rs]: Sure! Fixed.

**Review by muursh — APPROVED** [review]
Fine by me

File: p2p/src/net/mock/types.rs [hunk 2]
```diff
@@ -203,7 +203,7 @@ pub enum PeerEvent {
 }
 
 /// Events sent by the mock backend to peers
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```
Comment (TheQuantumPhysicist) [inline p2p/src/net/mock/types.rs:206]: While this is a mock and doesn't really matter, in production code, cloning the message is not really a good idea. Consider that a message can be a whole block, like 1 MB of data. If the logic requires having multiple copies floating around, then an Rc or Arc is the right way to go.

**Review by TheQuantumPhysicist — APPROVED** [review]

**Review by iljakuklic — APPROVED** [review]

**Pull request merged**